### DISCUSSION TITLE
A huge THANK YOU!

### DISCUSSION THREAD

**Discussion by notpron** [philippoo66/optolink-splitter#8] [General]
Hey @philippoo66,

You don't know me, but I've been using the optolink-splitter for the last 2 weeks. I've been working on adding my Vitoligno 300P to Home Assistant using vcontrold and vclient. It's been quite a learning curve. Three weeks ago, everything was mostly up and running. It was a big pain to set up and figure out addresses, then hooking it up to mqtt, adding robust systemd, etc. But what I kept running into was the speed of vcontrold, it was very slow.

Then I ran into one of your comments on the openv repo and ended up here. It took me a couple of evenings to migrate over, but I'm so, so very happy to have made the switch. You're solution requires a lot less complexity on my pi. Complexity that will lead to less maintenance down the road. And I'm still not sure what the difference is with vcontrold, but your setup is much faster. I'm guessing my config of vcontrold was using a lot of extra waits than necessary, I'm not sure. Using the address files in the other repo, I've also managed to control a lot more than I'd setup in the old configuration.

In all, a very positive experience. I hope I can contribute at some point.
Once more: THANK YOU!

**Comment by philippoo66** [comment]
hey @notpron 

thank you very much for your very positive feedback!! I'm really glad to hear that it works as easy as intended also for other people and I'm glad to hear that slowly the number of useres may rise!

This project started because I require the hot water temperature in my home automation since I intend to load it from PV if there is overage. So the MQTT aspect was born. But I did not want to lose Viessdata since I got so much used to it, so the TCP aspect came in. For both the polling list seemed to be a good thing to have. At the time I started I was aware of two people looking for the parallel use of Vitoconnect and local stuff, so I added another task to handle this....

In the end it was a good chunk of work - too much for only three people making use of it... ;-)

Whenever you have some idea of making something easier or adding some usefull feature or find a bug - you are very welcome to let me know!

Still I'm wondering if it would make sense to add a 'plain text datapoints table' to hold the address, data lenght, scale and signed to allow MQTT/TCP commands like `read;AmbientTempSensor` instead of `read;0x800;2;0.1;True` 

What do you think? Is it worth the effort (also the tables needs to get raised for all the different devices) or is it in the end that the tasks are usually carried out by scripts/machines and it doesn't matter if the 'table' is 'done' in the scripts or explicitly?

thank you & greetings!
Phil
